apps/Gruntfile.js

```diff
@@ -14,22 +18,66 @@ function log(str) {
 }
 
 module.exports = function (grunt) {
+  var locals = yaml.safeLoad(fs.readFileSync(path.join(__dirname, '..', 'locals.yml')));
+  var newrelicLicenseKey = locals.new_relic_license_key;
+  if (!newrelicLicenseKey) {
+    // we will skip logging to new relic.
+    console.log(chalk.yellow(
+        "\n\nAdd new_relice_license_key to your locals.yml file to\n" +
+          "have your build times logged to new relic. Talk to paul@code.org\n" +
+          "for more information.\n\n"
+    ));
+  }
+
   var email = 'unknown';
   try {
     email = execSync('git config --get user.email').toString().trim();
   } catch (e) {
     // I guess we are not in a git checkout, or don't have git installed.
   }
   timeGrunt(grunt, false, function (stats, done) {
+    var logFilePath = path.join(__dirname, 'build-times.log');
     try {
       fs.appendFileSync(
-        path.join(__dirname, 'build-times.log'),
+        logFilePath,
         JSON.stringify([new Date().toString(), email, stats])+'\n'
       );
     } catch (e) {
       log("failed to write to build-times.log file: "+e);
     }
-    done();
+    if (!newrelicLicenseKey) {
+      // we will skip logging to new relic.
+      done();
+      return;
+    }
+    var lineReader = readline.createInterface({
+      input: fs.createReadStream(logFilePath)
+    });
+    var dataToLog = [];
+    var lineIndex = 0;
+    lineReader.on('line', function (line) {
+      var data = JSON.parse(line);
+      var timestamp = Math.floor(new Date(data[0]).getTime()/1000); // seconds since epoch
+      var email = data[1];
+      var stats = data[2];
+      var totalTime = 0;
+      for (var i = 0; i < stats.length; i++) {
+        var task = stats[i][0];
+        var time = stats[i][1];
+        totalTime += time;
+      }
+      dataToLog.push({totalTime: totalTime, email: email, logTimestamp: timestamp});
```

Comment:
This looks like it will create a similar-looking log entry for any grunt command. If the goal is to measure build time, I think we will want at least enough information to distinguish between `grunt build` and `grunt test`. Perhaps it would make sense to include or sum certain task names?